build_projects/dotnet-cli-build/GenerateGuidFromName.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Runtime.InteropServices;
+using System.Net.Http;
+using Microsoft.Build.Utilities;
+using Microsoft.DotNet.Cli.Build.Framework;
+using Microsoft.DotNet.InternalAbstractions;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System.Security.Cryptography;
+
+using static Microsoft.DotNet.Cli.Build.Framework.BuildHelpers;
+
+namespace Microsoft.DotNet.Cli.Build
+{
+    public class GenerateGuidFromName : Task
+    {
+        [Required]
+        public string Name { get; set; }
+
+        [Output]
+        public string OutputGuid { get; set; }
+
+        public override bool Execute()
+        {
+            OutputGuid = GenerateGuid(Name).ToString();
```

Comment:
Our existing code does "ToUpper()" on this string, do we want it in the MSBuild world too?